src/Components/WebAssembly/testassets/ThreadingApp/Pages/FetchData.razor

```diff
@@ -0,0 +1,77 @@
+﻿@page "/fetchdata"
+@page "/fetchdata/{StartDate:datetime}"
+@inject HttpClient Http
+
+<h1>Weather forecast</h1>
+
+<p>This component demonstrates fetching data from the server.</p>
+
+@if (forecasts == null)
+{
+    <p><em>Loading...</em></p>
+}
+else
+{
+    <table class='table'>
+        <thead>
+            <tr>
+                <th>Date</th>
+                <th>Temp. (C)</th>
+                <th>Temp. (F)</th>
+                <th>Summary</th>
+            </tr>
+        </thead>
+        <tbody>
+            @foreach (var forecast in forecasts)
+            {
+                <tr>
+                    <td>@forecast.DateFormatted</td>
+                    <td>@forecast.TemperatureC</td>
+                    <td>@forecast.TemperatureF</td>
+                    <td>@forecast.Summary</td>
+                </tr>
+            }
+        </tbody>
+    </table>
+    <p>
+        <a href="fetchdata/@startDate.AddDays(-5).ToString("yyyy-MM-dd")" class="btn btn-secondary float-left">
+            ◀ Previous
+        </a>
+        <a href="fetchdata/@startDate.AddDays(5).ToString("yyyy-MM-dd")" class="btn btn-secondary float-right">
+            Next ▶
+        </a>
+    </p>
+}
+
+@code {
+    [Parameter] public DateTime? StartDate { get; set; }
+
+    WeatherForecast[] forecasts;
+    DateTime startDate;
+
+    protected override async Task OnParametersSetAsync()
+    {
+        startDate = StartDate.GetValueOrDefault(DateTime.Now);
+
+        // send me to the thread pool, next line HTTP should work anyway
+        await Task.Delay(10).ConfigureAwait(false);
```

Comment:
This pattern is not supported in Blazor. When we see users do something like this, we tell them it's not supported. It makes the callback leave the Blazor sync context (on server and now on wasm) which we don't consider valid.

Reply:
I will make it more explicit threadpool task